sdk/storage/Azure.Storage.Common/src/Sas/AccountSasServices.cs

```diff
@@ -2,103 +2,72 @@
 // Licensed under the MIT License.
 
 using System;
-using System.ComponentModel;
 using System.Text;
 
 namespace Azure.Storage.Sas
 {
     /// <summary>
     /// Specifies the services accessible from an account level shared access
-    /// signature.  Use <see cref="ToString"/> to produce a value that can be
-    /// used for <see cref="AccountSasBuilder.Services"/>.
+    /// signature.
     /// </summary>
-    public struct AccountSasServices : IEquatable<AccountSasServices>
+    [Flags]
+    public enum AccountSasServices
     {
         /// <summary>
-        /// Gets a value indicating whether Azure Blob Storage resources are
+        /// Indicates whether Azure Blob Storage resources are
         /// accessible from the shared access signature.
         /// </summary>
-        public bool Blobs { get; set; }
+        Blobs = 1,
 
         /// <summary>
-        /// Gets a value indicating whether Azure Queue Storage resources are
+        /// Indicates whether Azure Queue Storage resources are
         /// accessible from the shared access signature.
         /// </summary>
-        public bool Queues { get; set; }
+        Queues = 2,
 
         /// <summary>
-        /// Gets a value indicating whether Azure File Storage resources are
+        /// Indicates whether Azure File Storage resources are
         /// accessible from the shared access signature.
         /// </summary>
-        public bool Files { get; set; }
+        Files = 4,
 
+        /// <summary>
+        /// Indicates all services are accessible from the shared
+        /// access signature.
+        /// </summary>
+        All = ~0
+    }
+
+    /// <summary>
+    /// Extension methods for AccountSasServices enum
+    /// </summary>
+    internal static partial class SasExtensions
+    {
         /// <summary>
         /// Creates a string representing which services can be used for
         /// <see cref="AccountSasBuilder.Services"/>.
         /// </summary>
         /// <returns>
         /// A string representing which services are allowed.
         /// </returns>
-        public override string ToString()
+        internal static string ToServicesString(this AccountSasServices services)
```

Comment:
Can we call all of these `ToPermissionsString` instead?

Reply:
Updated Account/Resources to use that name.